editor/editor_help.cpp

```diff
@@ -2618,20 +2618,25 @@ static void _add_text_to_rt(const String &p_bbcode, RichTextLabel *p_rt, Control
 			p_rt->set_cell_padding(Rect2(10 * EDSCALE, 10 * EDSCALE, 10 * EDSCALE, 10 * EDSCALE));
 			p_rt->push_color(code_dark_color);
 
-			if (lang.is_empty() || lang == "gdscript") {
+			bool highlighted = false;
+
 #ifdef MODULE_GDSCRIPT_ENABLED
+			if (!highlighted && lang == "gdscript") {
 				EditorHelpHighlighter::get_singleton()->highlight(p_rt, EditorHelpHighlighter::LANGUAGE_GDSCRIPT, codeblock_text, is_native);
-#else
-				p_rt->add_text(codeblock_text);
+				highlighted = true;
+			}
```

Comment:
This changes behavior, the previous code would handle `lang.is_empty()` by inferring GDScript by default. I believe we want to preserve this behavior, instead plain-text highlighting needs to be set explicitly with `lang == "text"` (see e.g. #89316).

Reply:
You can restore the intended behavior with `if (!highlighted && (lang.is_empty() || lang == "gdscript"))`.

Reply:
The fact is that in most cases `[codeblock]` uses GDScript, so I made it the default. Same with RST, the default `::` is highlighted as GDScript, we need explicit `.. code::` to remove the highlighting.

Reply:
Yes, highlighting GDScript by default is very, very favourable.